src/compiler/checker.ts

```diff
@@ -4324,7 +4324,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     combineValueAndTypeSymbols(symbolFromVariable, symbolFromModule) :
                     symbolFromModule || symbolFromVariable;
                 if (!symbol) {
-                    errorNoModuleMemberSymbol(moduleSymbol, targetSymbol, node, name);
+                    errorNoModuleMemberSymbol(moduleSymbol, targetSymbol, getSourceFileOfNode(node), name);

```

Comment:
This relies on the `printer` selection in `symbolToStringWorker`. While the containing function also accepts `node.kind === SyntaxKind.VariableDeclaration`, such a node doesn't reach this line in any test case so I'm currently not 100% sure if I can just call `getSourceFileOfNode` unconditionally here. I think it's OK for imports and exports.

Reply:
Misrepresenting the location that we’re going to scope a symbol reference from sounds like a bad idea—the original fix in #39463 that you’re building on here feels a little too magically indirect to begin with to me. Is there any way you see to make the printer escaping behavior more directly dependent on whether a module specifier is being printed?